risc0/zkvm/src/host/client/prove/default.rs

```diff
@@ -0,0 +1,163 @@
+// Copyright 2025 RISC Zero, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::{
+    cell::RefCell,
+    fs::File,
+    io::{Read, Write},
+    os::{
+        fd::{FromRawFd, IntoRawFd},
+        unix::net::UnixStream,
+    },
+    path::Path,
+    process::{Child, Command},
+    sync::Arc,
+};
+
+use anyhow::{bail, Context as _, Error, Result};
+
+use crate::{
+    rpc::{JobInfo, JobStatus, ProofRequest},
+    ExecutorEnv, ProveInfo, Receipt, SessionInfo, SessionStats, VerifierContext,
+};
+
+use super::{Executor, Prover, ProverOpts};
+
+/// TODO
+pub struct DefaultProver {
+    child: Child,
+    socket: RefCell<UnixStream>,
+}
+
+/// An implementation of a [Prover] that runs proof workloads via local `r0vm` cluster.
+impl DefaultProver {
+    /// Construct a [DefaultProver].
+    pub fn new<P: AsRef<Path>>(r0vm_path: P) -> Result<Self> {
+        let r0vm_path = r0vm_path.as_ref();
+
+        let (parent_socket, child_socket) = UnixStream::pair()?;
+        let mut cmd = Command::new(r0vm_path);
+        let child_fd = child_socket.into_raw_fd();
+        let child_file = unsafe { File::from_raw_fd(child_fd) };
```

Comment:
i think you can just do `Stdin::from_raw_fd` here without going to a `File`, the `stdin` call seems to convert it to that type anyway.

Reply:
There's a conversion to a `Stdio` and only from `File` (and other types) are supported, but I couldn't get going direction from `UnixStream` to work.

Reply:
I had I typo with what I was saying, I guess I meant you could use this function https://doc.rust-lang.org/std/process/struct.Stdio.html#impl-FromRawFd-for-Stdio

Reply:
OK, I fixed this.